NMS-14351: Add terms to glossary.adoc (#4899)
OpenNMS/opennms: docs/modules/reference/pages/glossary.adoc

xref:operation:alarms/introduction.adoc[Alarms]:: Indicators for problems occurring on entities that {page-component-title} monitors.

https://docs.opennms.com/alec/latest/[Architecture for Learning Enabled Correlation (ALEC)]:: An OpenNMS framework that uses artificial intelligence and machine learning to provide alarm triage.
ALEC groups related alarms so you can better manage the more complex situations they might represent.

xref:operation:application-perspective-monitoring/introduction.adoc[Application Perspective Monitoring (APM)]:: A method that uses the Minion infrastructure to monitor a service’s availability from different perspectives to help identify if nodes are down for everyone or if an outage affects only certain sites.
When a service is not responsive an outage will be generated with the corresponding perspective that sees the outage.

> Asset:: Any data, device, or other component that is monitored by {page-component-title}.

Reviewer: Devices monitored are referred to as "nodes".
```suggestion
Asset data:: In the context of {page-component-title}, asset data is a set of optional fields that can be associated with a node, similar to metadata, to provide additional details.
Examples of asset data include physical location and hardware specifications.
```

Reviewer: Looks good.

Bean Scripting Framework (BSF):: A framework that allows scripting to be used in Java code via a set of Java classes.
The {page-component-title} xref:reference:provisioning/detectors/BsfDetector.adoc[BSF Detector] runs arbitrary BSF scripts to determine the existence of a service, and the xref:reference:service-assurance/monitors/BSFMonitor.adoc[BSF Monitor] uses scripts to determine the status of a service.

> xref:operation:bsm/introduction.adoc[Business Service Monitoring (BSM)]:: A {page-component-title} feature that monitors high-level business Service Monitor contexts and quickly identifies the most critical problems affecting them.

Reviewer: Just realized this is a circular definition.
```suggestion
xref:operation:bsm/introduction.adoc[Business Service Monitoring (BSM)]:: A {page-component-title} feature that allows you to alert on issues affecting complex business services that span multiple nodes.
```

> xref:operation:bsm/introduction.adoc[Business Service Monitoring (BSM)]:: A {page-component-title} feature that monitors high-level business Service Monitor contexts and quickly identifies the most critical problems affecting them.

Author: ```suggestion
xref:operation:bsm/introduction.adoc[Business Service Monitoring (BSM)]:: A {page-component-title} feature that allows you to monitor and generate alerts for issues affecting complex business services spanning multiple nodes.
```

> xref:operation:bsm/introduction.adoc[Business Service Monitoring (BSM)]:: A {page-component-title} feature that monitors high-level business Service Monitor contexts and quickly identifies the most critical problems affecting them.

Reviewer: looks good

Cassandra:: An open-source distributed/scale-out NoSQL database designed to hold large amounts of structured or unstructured data across many servers.
The optional xref:deployment:time-series-storage/newts/introduction.adoc[Newts] storage strategy requires a Cassandra instance as back-end storage.

Categories:: User-defined groups of entities on {page-component-title}-monitored networks.

> They allow you to define a matrix for aggregating monitoring results.

Reviewer 2: This is a NH thing. They prefer "lets" instead of "allows". I haven't been updating it, but it is one of the things we do need to watch.

> They allow you to define a matrix for aggregating monitoring results.

Author: Thanks for letting me know! I'll keep this in mind. :)

Discourse:: An open-source forum platform that hosts https://opennms.discourse.group/[OpenNMS's community forums].
There, users can find knowledge base articles, support, announcements, and general discussion about all things OpenNMS.

xref:operation:provisioning/introduction.adoc#discovery-auto[Discovery]:: A process that uses ICMP to find network devices and services in your network and then automatically adds them to monitoring.

> https://docs.docker.com/[Docker]:: An open-source container virtualization service that allows you to deploy application-level services without worrying about operating system maintenance.

Reviewer 2: ```suggestion
https://docs.docker.com/[Docker]:: An open-source container virtualization service that lets you deploy application-level services without worrying about operating system maintenance.
```

{page-component-title} components may optionally be deployed as Docker containers.

Dominion:: The service on an OpenNMS core that controls the operation of Minions.
(See xref:reference:configuration/minion-confd/minion-confd.adoc#dominion[Configuring Minion via Confd].)

> Drools:: A Business Rules Management System solution that supports the Java Rules Engine API standard.

Reviewer 2: ```suggestion
Drools:: A business rules management system that supports the Java Rules Engine API standard.
```
I'm not clear on if it's a solution that helps with business rules management systems, or if it is a system that helps with business rules management.

Author: I mostly grabbed this from the Wikipedia page, where the first sentence is,
> Drools is a business rule management system (BRMS) with a forward and backward chaining inference based rules engine, more correctly known as a production rule system, using an enhanced implementation of the Rete algorithm.

> It is used in the {page-component-title} Alarmd daemon to provide a more robust infrastructure for workflow and problem state management.

Reviewer 2: ```suggestion
It helps provide a more robust infrastructure for workflow and problem state management in the {page-component-title} Alarmd daemon.
```
Unless it is Alarmd that is providing the more robust infrastructure.

Geolocation:: The actual physical location of a node monitored by {page-component-title}.
Nodes can be manually assigned a location, or the location can be automatically determined using the xref:reference:provisioning/adapters/geoip.adoc[GeoIP Provisioning Adapter].

https://grafana.com/[Grafana]:: An open-source analysis and visualization web application.
It connects to data sources and generates dashboards with charts, graphs, and alerts.

> https://docs.opennms.com/helm/latest[OpenNMS HELM] is a Grafana plugin that allows users to create customized dashboards with data from {page-component-title}.

Reviewer 2: ```suggestion
https://docs.opennms.com/helm/latest[OpenNMS HELM] is a Grafana plugin that lets users create customized dashboards with data from {page-component-title}.
```

xref:operation:admin/webui/opsboard/dashlet/ksc.adoc#ksc[KSC reports]:: Key SNMP customized (KSC) reports provide a way to generate prefabricated graphical views of collected data.
They let you display data from different devices and sources (SNMP, ICMP, HTTP) on one page.

https://kubernetes.io/[Kubernetes]:: An open-source container orchestration system for automating software deployments, scaling, and management.
Originally designed by Google, the Cloud Native Computing Foundation now maintains it.

Location:: Also referred to as "monitoring location" or "Minion location".
Nodes assigned to a monitoring location will be monitored by Minions assigned to the same location.
This is distinct from geolocation, which is the actual physical location of the node.

> Management Information Base (MIB):: A database used for managing entities on a network.
> Used in {page-component-title} to facilitate Transparent Bridge Discovery.
> (See xref:operation:topology/enlinkd/layer-2/bridge-discovery.adoc[Bridge Discovery].)

Reviewer: MIBs are text files that describe SNMP data.  There is a tool to compile them into XML config files, though there isn't any documentation on that tool.
```suggestion
Management Information Base (MIB):: A file provided by a hardware or software vendor that describes SNMP objects provided by their products.
The MIB files can be processed to convert the definition into data collection and event objects for {page-component-title} to work with.
Many of the major vendor MIB definitions are included out of the box.
```

https://www.opennms.com/meridian/[Meridian]:: A subscription-based, optimized, and stable version of the OpenNMS Horizon platform that includes only the most stable and secure Horizon features.
It is distributed via annual release with monthly security patches.

Message broker:: When using Minions and Sentinels, a message broker is required for communication between servers.
{page-component-title} has an embedded ActiveMQ server available for smaller deployments.
You can swap the broker with a dedicated ActiveMQ, gRPC, or Kafaka cluster to provide scalability and load balancing.

> xref:development:minion/introduction.adoc#minion[Minion]:: An instance of the Karaf OSGi service that enables OpenNMS to monitor devices and services in locations that OpenNMS cannot reach.

Reviewer: Many operator users don't know what a "Karaf OSGi service" is, and don't really need to know unless they are writing code.
```suggestion
xref:development:minion/introduction.adoc#minion[Minion]:: A distributed component that enables OpenNMS to monitor devices and services in locations that OpenNMS cannot reach.
```

Reviewer 2: We should update the cloud docs to reflect this.

xref:operation:admin/webui/opsboard/introduction.adoc[Operator board]:: Customizable dashboard to visualize monitoring information.

Outage:: Also known as _downtime_, this is a period of time that a service is unavailable or offline.

> It can be caused by systems or communications failures, or it can be planned as part of routine maintenance.

Reviewer 2: ```suggestion
It can be caused by systems or communications failures, or planned as part of routine maintenance.
```

xref:operation:topology/introduction.adoc[Topology]:: A description of a network's elements (devices, services) and the connections among them.

xref:operation:events/sources/snmp-traps.adoc[Traps]:: Event triggers generated by SNMP-capable devices on the network and sent to the {page-component-title} Trapd service daemon.

Unique Event Identifier (UEI):: A string that uniquely identifies an event's type.

> UEIs must begin with the string `uei..`.

Reviewer: ```suggestion
UEIs must begin with the string `uei.`.
```
